src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private MyLinkedList.Node<T> first;
+    private MyLinkedList.Node<T> last;
+
+    MyLinkedList.Node<T> node(int index) {
+        Node<T> foundedNode;
+        if (index < (size / 2)) {
+            foundedNode = first;
+            for (int i = 0; i < index; i++) {
+                foundedNode = foundedNode.next;
+            }
+        } else {
+            foundedNode = last;
+            for (int i = size - 1; i > index; i--) {
+                foundedNode = foundedNode.prev;
+            }
+        }
+        return foundedNode;
+    }
+
+    private void checkElementIndex(int index) {
+        if (!(index >= 0 && index < size)) {
+            throw new IndexOutOfBoundsException("Index: " + index + " is more then size " + size);
+        }
+    }
+
+    private void checkPositionIndex(int index) {
+        if (!(index >= 0 && index <= size)) {
+            throw new IndexOutOfBoundsException("Index: " + index + " is more then size " + size);
+        }
+    }
+
+    void unlink(Node<T> removedNode) {
+        final MyLinkedList.Node<T> next = removedNode.next;
+        final MyLinkedList.Node<T> prev = removedNode.prev;
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+            removedNode.prev = null;
+        }
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            removedNode.next = null;
+        }
+        removedNode.item = null;
+        size--;
+    }
+
+    void linkLast(T element) {
+        final MyLinkedList.Node<T> newNode = new MyLinkedList.Node<>(last, element, null);
+        if (last == null) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
+    }
+
+    void linkBefore(T element, MyLinkedList.Node<T> afterNode) {
+        final MyLinkedList.Node<T> prev = afterNode.prev;
+        final MyLinkedList.Node<T> newNode = new MyLinkedList.Node<>(prev, element, afterNode);
+        if (prev == null) {
+            first = newNode;
+        } else {
+            prev.next = newNode;
+        }
+        afterNode.prev = newNode;
+        size++;
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
```

Comment:
Wouldn't it be better to just write what is written in the linkLast method here? And then use the add method where needed